### PR TITLE
Deprecate kubectl scale job

### PR DESCRIPTION
**Pull request by soltysh**
**What this PR does / why we need it**:
With the generic scaler (https://github.com/kubernetes/kubernetes/pull/58298) the only problem is job and as discussed in https://github.com/kubernetes/kubernetes/pull/58468#issuecomment-360794229 and during SIG CLI we've agreed that scaling jobs was a mistake we need to revert. 
This PR deprecates scale command for jobs, only. 

/assign @deads2k @pwittrock 

**Release note**:
```release-note
Deprecate kubectl scale jobs (only jobs). 
```

### CODE REVIEW

**Comment by deads2k** [comment]
This matches the discussion in sig-cli.

/lgtm

File: pkg/kubectl/cmd/scale.go [hunk 1]
```diff
@@ -146,6 +147,10 @@ func RunScale(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 		}
 
 		mapping := info.ResourceMapping()
+		if mapping.Resource == "jobs" {
+			glog.Warningf("%s scale job is DEPRECATED and will be removed in a future version.", cmd.Parent().Name())
```
Comment (deads2k) [inline pkg/kubectl/cmd/scale.go]: we normally use `fmt.fprintf(stderr` to avoid irrelevant line numbers.

**Comment by deads2k** [comment]
/hold

fix up the printing

File: pkg/kubectl/cmd/scale.go (hunk 1 above)
Comment (soltysh) [reply pkg/kubectl/cmd/scale.go]: That's what I had originally, but then I found [this cherry](https://github.com/kubernetes/kubernetes/blob/ecc5eb67d965295db95ba2df5f3d3ff43a258a05/pkg/kubectl/cmd/cmd.go#L358-L360). Although I'm fine fixing the other one as well to use `fmt.Fprintf`

**Comment by soltysh** [comment]
@deads2k fixed the printing and updated `printDeprecationWarning` accordingly

**Comment by soltysh** [comment]
The printing was fixed, I'm guessing David for got to :
/hold cancel

**Pull request merged**